apps/condo/domains/billing/utils/countrySpecificQRCodeParsers/ru.qrCodeParser.js

```diff
@@ -14,26 +17,26 @@ const { get } = require('lodash')
 
 /**
  * @param {string} qrStr The QR code string got from the picture
+ * @param {function} logger ValidateQRCodeService logger
  * @return {TRUQRCodeFields}
  */
 function parseRUReceiptQRCode (qrStr) {
-    const matches = /^ST(?<version>\d{4})(?<encodingTag>\d)\|(?<requisitesStr>.*)$/g.exec(qrStr)
+    const buffer = Buffer.from(qrStr, 'base64')
+    const detectedEncoding = detectEncoding(buffer)
+    const decodedRequisitesStr = convertEncoding(buffer, detectedEncoding)
+
+    const matches = /^ST(?<version>\d{4})(?<encodingTag>\d)\|(?<requisitesStr>.*)$/g.exec(decodedRequisitesStr)
 
     if (!matches) {
+        logger.error({ msg:'Error qr-code parsing', data: { qrStr, decodedRequisitesStr, detectedEncoding }, step: 'decoded' })
```

Comment:
What is `step`? Is it really necessary with single value?
Please consider to remove this field or move it to data to prevent log indexing by this uncommon field.